model/model.js

```diff
@@ -0,0 +1,61 @@
+const LOCAL_STORAGE_TODOS_KEY = 'LOCAL_STORAGE_TODOS_KEY';
+const todos = JSON.parse(localStorage.getItem(LOCAL_STORAGE_TODOS_KEY)) || {};
+
+const LOCAL_STORAGE_TODO_CATEGORIES_KEY = 'LOCAL_STORAGE_TODO_CATEGORIES_KEY';
+const todoCategories = JSON.parse(localStorage.getItem(LOCAL_STORAGE_TODO_CATEGORIES_KEY)) || [];
+
+function saveTodos() {
+    localStorage.setItem(LOCAL_STORAGE_TODOS_KEY, JSON.stringify(todos));
+}
+function saveTodoCategories() {
+    localStorage.setItem(LOCAL_STORAGE_TODO_CATEGORIES_KEY, JSON.stringify(todoCategories));
+}
+
+class createObject {
+    #_id; #_todoCategory; #_value; #_isComplete; #_isPin;
+
+    constructor(id, todoCategory, value, isComplete, isPin) {
+        this.#_id = id;
+        this.#_todoCategory = todoCategory;
+        this.#_value = value;
+        this.#_isComplete = isComplete;
+        this.#_isPin = isPin;
+    }
+
+    get id() {
```

Comment:
use normal functions only and rename functions accordingly.
thus we don't need to do deep clone of objects for getting these properties